datafusion/src/sql/utils.rs

```diff
@@ -390,6 +391,30 @@ pub(crate) fn extract_aliases(exprs: &[Expr]) -> HashMap<String, Expr> {
         .collect::<HashMap<String, Expr>>()
 }
 
+pub(crate) fn resolve_positions_to_exprs(
+    expr: &Expr,
+    select_exprs: &[Expr],
+) -> Result<Expr> {
+    match expr {
+        Expr::Literal(position) => match position {
+            ScalarValue::Int64(Some(pos)) => {
```

Comment:
@Dandandan do you think we should handle other Integer and unsigned integer types?

if so, do you know how we can make the code DRY? since following raise type error
```
Expr::Literal(position) => match position {
    |                                          -------- this expression has type `&scalar::ScalarValue`
400 |             ScalarValue::UInt64(Some(pos)) | ScalarValue::Int64(Some(pos)) => {
    |                                      ---                             ^^^ expected `u64`, found `i64`
    |                                      |
    |                                      first introduced with type `&u64` here
```